### DISCUSSION TITLE
Issues with official website cases (parse the css transform of the canvas)

### DISCUSSION THREAD

**Discussion by guda-art** [fabricjs/fabric.js#10345] [Q&A] [answered]
![image](https://github.com/user-attachments/assets/8e364dce-e122-495a-ae9b-7cb094fc0509)
Sorry, I know this is a side effect caused by other reasons, but after adding 'transform: rotate', **I can't select the control point** anymore. I cried

**Comment by asturur** [comment]
We don't parse the css transform of the canvas indeed. We have just an approximate scaling if there is a difference between with and actual width.

**Reply by asturur** [reply]
What would be the reason to css rotate the canvas?

**Reply by guda-art** [reply]
I abstracted a behavior from my code and added it to the external container of the 'canvas' in this case, causing the mouse events on the canvas to shift. This means that when using canvas through fabric, we need to ensure that the canvas is not affected by the side effects of transform
thanks for you answer

**Reply by guda-art** [reply]
parse the css transform of the canvas,  Can I do this on my own? In theory, it seems feasible, but I'm not sure about its feasibility 😄

**Reply by asturur** [reply, marked as answer]
The css transform is so complicated, unless you can get it ready made by some browser api that can parse it for you from the canvas up to the dom including everything that is in the middle.
Once you know which is the transform applied to the canvas, you can add it to the viewport transform i suppose

**Reply by guda-art** [reply]
thankyou  good
